core/util/src/mill/util/Jvm.scala

```diff
@@ -241,7 +260,7 @@ object Jvm {
    */
   def withClassLoader[T](
       classPath: Iterable[os.Path],
-      parent: ClassLoader = null,
+      parent: ClassLoader = Thread.currentThread().getContextClassLoader,
```

Comment:
We should remove the default. Using the context classloader should be a conscious decision at the use site.